src/javascript/JContent/ContentRoute/ContentLayout/FilesGrid/FileCard/FileCard.jsx

```diff
@@ -117,6 +124,9 @@ export const FileCard = ({
     let columnNumber = (index % 2) + 1;
     let encodedPath = node.path.replace(/[^/]/g, encodeURIComponent);
     let isPdf = node.children ? node.children.nodes.filter(node => node.mimeType.value === 'application/pdf').length !== 0 : false;
+
+    let showSubNodes = node.primaryNodeType.name !== 'jnt:page' && node.subNodes && node.subNodes.pageInfo.totalCount > 0;
```

Comment:
do we need to add null check for pageInfo in there? alternatively might be better to use optional chaining e.g. `node?.subNodes?.pageInfo?.totalCount > 0`. Also make sure it's returning boolean and not falsey value as falsey might still render

Reply:
Oh cool never knew that awesome

Reply:
if showSubNodes is null or undefined then I believe it's fine. but for some other falsey values (e.g. 0 or Nan), those values can still get rendered https://dev.to/aviskarkc10/jsx-gotcha-when-conditionally-rendering-using-179j

In general, I'd just try to avoid falsey values and just convert to straight up boolean (using !!) or ternary operator.

Reply:
Gotcha the example make sense
Kindly take another look 
As with the optional chaining it will return undefined if the value does not exist (which is good in our case)
Plus other values will be either true or false (believe we are covered)